src/Accelerators/NNPA/Conversion/ONNXToZHigh/RewriteONNXForZHigh.cpp

```diff
@@ -224,9 +224,13 @@ bool canInferencePadsForNNPAConv(ONNXConvOp op) {
   ONNXConvOpAdaptor operandAdaptor = ONNXConvOpAdaptor(op);
   ONNXConvOpShapeHelper shapeHelper(&op);
   assert(succeeded(shapeHelper.computeShape(operandAdaptor)));
+  RankedTensorType inputType = op.X().getType().cast<RankedTensorType>();
+  ArrayRef<int64_t> inputShape = inputType.getShape();
   return (shapeHelper.pads.size() == 4) &&
-         (llvm::all_of(
-             shapeHelper.pads, [](IndexExpr val) { return val.isLiteral(); }));
+         (llvm::all_of(shapeHelper.pads,
+             [](IndexExpr val) { return val.isLiteral(); })) &&
+         (op.auto_pad() != "VALID") && (inputShape.size() == 4) &&
+         (inputShape[1] > 0) && (inputShape[2] > 0) && (inputShape[3] > 0);
```

Comment:
Could you split this complex condition into small ones and add a comment for each, like
```C
// comment
if ()
  return false;
// comment
if ()
  return false;
// Otherwise, the op is legal.
return true;
```
It makes the code easy to read and debug. Thanks!

Reply:
Fixed. Thank you.